src/NServiceBus.RavenDB/Outbox/OutboxPersister.cs

```diff
@@ -36,10 +34,15 @@ public OutboxPersister(IDocumentStore documentStore, EndpointName endpointName)
                 return default(OutboxMessage);
             }
 
-            var operations = new List<TransportOperation>();
+            var operations = new TransportOperation[result.TransportOperations.Length];
             if (!result.Dispatched)
             {
-                operations.AddRange(result.TransportOperations.Select(t => new TransportOperation(t.MessageId, t.Options, t.Message, t.Headers)));
+                var index = 0;
+                foreach (var operation in result.TransportOperations)
```

Comment:
List and array have a struct enumerator. So we are fine. And we need access to the current element so a foreach is more convenient than using the indexer all over the place